spotipy/json_types.py

```diff
@@ -0,0 +1,573 @@
+import sys
+from typing import Any, Generic, List, Mapping, TypeVar, overload
+
+if sys.version_info >= (3, 8):
+    from typing import Literal, TypedDict
+else:
+    from typing_extensions import Literal, TypedDict
+
+_T = TypeVar("_T", covariant=True)
+
+
+TokenInfo = TypedDict("Token", {
+    "access_token": str,
+    "token_type": str,
+    "scope": str,
+    "expires_in": int,
+    "refresh_token": str,
+    "expires_at": int
+})
```

Comment:
`scope` and `refresh_token` will not be present for the client credentials flow, and so they should be optional.

Reply:
In the future we could look into creating a new type if possible, but probably not a worry now

Reply:
I discussed this [here](https://github.com/plamere/spotipy/issues/439#issuecomment-872464912). We can either:
- Ignore the fact that they will not be present
- Create a class `TokenInfo`
- Remove this, use `Dict[str, Any]`, and wait until `TypeDict` supports optional keys (or just forget it)